File: src/main/java/InvertBinaryTree/yunyuntop/InvertBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if(root==null){
            return root;
        }
        TreeNode tmp;
        tmp=root.right;
        root.right = root.left;
        root.left = tmp;
        if(root.left!=null){
            invertTree(root.left);
        }
        if(root.right!=null){
            invertTree(root.right);
        }
        return root;
    }
}

File: src/main/java/InvertBinaryTree/BBIGBOSS/InvertBinaryTree.cpp
//
// Created by BBIGBOSS on 2019-06-15.
//

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution
{
    public:

    TreeNode* invertTree(TreeNode* root)
    {
        if (root)
        {
            swap(root->left, root->right);

            invertTree(root->left);
            invertTree(root->right);
        }

        return root;
    }
};